EssentialApp/EssentialAppTests/FeedAcceptanceTests.swift

```diff
@@ -51,6 +51,25 @@ class FeedAcceptanceTests: XCTestCase {
 		
 		XCTAssertNotNil(store.feedCache, "Expected to keep non-expired cache")
 	}
+    
+    func test_onSelectImage_displaysImageComments() {
+        let feed = launch(httpClient: .online(response), store: .empty)
+        
+        feed.simulateTapOnImage(at: 0)
+        RunLoop.current.run(until: Date())
+        
+        XCTAssertNotNil(feed.navigationController?.topViewController as? FeedImageCommentsViewController)
+        
+        let comments = feed.navigationController?.topViewController as? FeedImageCommentsViewController
```

Comment:
You can define the `let comments` before the `XCTAssertNotNil`, so you avoid duplicating the code.

```suggestion
        let comments = feed.navigationController?.topViewController as? FeedImageCommentsViewController

        XCTAssertNotNil(comments)
```

You can also do it all in one step with `XCTUnwrap`  (you need to add `throws` to the test method signature):

```suggestion        
        let comments = try XCTUnwrap(feed.navigationController?.topViewController as? FeedImageCommentsViewController)
```